app/controllers/foreman_salt/concerns/hostgroups_controller_extensions.rb

```diff
@@ -8,7 +8,7 @@ module HostgroupsControllerExtensions
       end
 
       def salt_environment_selected
-        @hostgroup = Hostgroup.authorized(:view_hostgroups, Hostgroup).find_by_id(params[:hostgroup_id]) || Hostgroup.new(params[:hostgroup])
+        @hostgroup = Hostgroup.authorized(:view_hostgroups, Hostgroup).find_by(:id => params[:hostgroup_id]) || Hostgroup.new(params[:hostgroup])
```

Comment:
`find_by` methods ought to work on Rails 4. It's just `find_or_create_by` and `find_or_initialize_by` that were removed